geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/iterators/BaseAggregatingIterator.scala

```diff
@@ -82,3 +83,21 @@ abstract class BaseAggregatingIterator[T <: AggregatingScan.Result]
   override def deepCopy(env: IteratorEnvironment): SortedKeyValueIterator[Key, Value] =
     throw new NotImplementedError()
 }
+
+object BaseAggregatingIterator {
+
+  private class BatchScanCallback extends AggregateCallback {
+
+    private var bytes: Array[Byte] = _
+
+    override def batch(bytes: Array[Byte]): Boolean = {
+      this.bytes = bytes
+      false // we want to stop scanning and return the batch
+    }
+
+    // we always keep scanning and rely on client connections to stop the scan
+    override def partial(bytes: => Array[Byte]): Boolean = true
```

Comment:
Ok, for this class.... if partial is called, what happens to 'bytes'?  When would it actually be executed to continue the scan?

Reply:
bytes is lazily evaluated, so in this case it will never be calculated, and then we return true so the scan continues on.

Reply:
Ok....  I thought calling 'bytes' would be what continues the scan...

Reply:
i tried to explain it in the scala docs, but possibly not very well: https://github.com/locationtech/geomesa/pull/2463/files#diff-2feb8b7c8aaa8331044930a8332e7291R282
returning true/false is what tells it to continue the scan, the bytes are provided in case the caller wants to stop the scan and return any partial results, otherwise they will be bundled in the next batch callback.